StatisticallySpeaking/StatisticallySpeaking.cpp

```diff
@@ -56,54 +53,57 @@ void StatisticallySpeaking::onEventMatchEnded(string eventName)
 							matchValues["Score"] = to_string(localPrimaryPlayer.GetMatchScore());
 							matchValues["MVP"] = to_string(localPrimaryPlayer.GetbMatchMVP());
 
-							SteamID steamIdentification;
-							steamIdentification.ID = gameWrapper->GetSteamID();
+							gameWrapper->SetTimeout([this, playlistId](GameWrapper* gameWrapper) {
+								getMMRAndSaveFile(playlistId);
+								}, 6.0f);
+						}
+					}
+				}
+			}
+		}
+	}
+}
 
-							gameWrapper->SetTimeout([this, steamIdentification, playlistId](GameWrapper* gameWrapper) {
-								matchValues["MMR"] = to_string((int)gameWrapper->GetMMRWrapper().GetPlayerMMR(steamIdentification, playlistId));
+void StatisticallySpeaking::getMMRAndSaveFile(int playlistId) {
+	auto playerId = gameWrapper->GetUniqueID();
 
-								string matchesFilePath = "./bakkesmod/data/StatisticallySpeaking-Matches-" + to_string(steamIdentification.ID) + ".csv";
+	if (gameWrapper->GetMMRWrapper().IsSynced(playerId, playlistId) &&
+		!gameWrapper->GetMMRWrapper().IsSyncing(playerId)){
 
-								if (!ifstream(matchesFilePath)) {
-									matchesFile.open(matchesFilePath, ios_base::app);
+		auto matchesFilePath = gameWrapper->GetDataFolder() / ("StatisticallySpeaking-Matches-" + playerId.str() + ".csv");
```

Comment:
`playerId.str()` has been returning either an Epic ID or Steam ID for me seemingly randomly after matches. This has resulted in my stats being split across 2 different CSV files.

Bakkes says to use the new `uniqueIdWrapper->GetIdString()` function instead which will be consistent. It "returns the id in the following format platform|accountId|splitScreenId" so the | characters will need to be replaced for the filename.

Reply:
Thank you. I've heard of that function before, but it was not there when I wrote this pull. I also noticed the ID mismatch happening from time to time. Hopes it will fix it.

I'll look at it.